src/components/CardInfo.tsx

```diff
@@ -0,0 +1,160 @@
+                        />
+                    </ModalHeader>
+
+                    <ModalCloseButton />
+                    <ModalBody pb={6}>
+                        <FormControl mt={4}>
+                            <FormLabel>Comments</FormLabel>
+                            <Textarea
+                                placeholder="Add comment..."
+                                value={addComment}
+                                onChange={(e) => {
+                                    setAddComment(e.target.value);
+                                }}
+                            />
+                            <Button
+                                colorScheme="blue"
+                                onClick={handleAddComment}
+                                marginTop={"2vh"}
+                            >
+                                Add
+                            </Button>
+                        </FormControl>
+                        <List spacing={3} marginTop={"4vh"}>
+                            {comments.map((com) => (
+                                <ListItem key={com.comment_id}>
+                                    <ListIcon
+                                        as={SunIcon}
+                                        color={"yellow.500"}
+                                    />
+
+                                    <Editable
+                                        value={com.text}
+                                        onChange={(e) => {
+                                            handleEditComment(
+                                                com.comment_id,
+                                                e
+                                            );
+                                        }}
+                                        onSubmit={(e) => {
+                                            handleSubmitComment(
+                                                com.comment_id,
+                                                e
+                                            );
+                                        }}
+                                    >
+                                        <EditablePreview />
+                                        <EditableTextarea />
+                                    </Editable>
+                                    <IconButton
+                                        icon={<DeleteIcon />}
+                                        colorScheme={"red"}
+                                        aria-label="delete button"
+                                        onClick={() => {
+                                            handleDeleteComment(com.comment_id);
+                                        }}
+                                    />
+                                </ListItem>
+                            ))}
+                        </List>
+                    </ModalBody>
+                </ModalContent>
+            </Modal>
+        </>
+    );
+}
```

Comment:
This is a pyramid of doom (google it). Very hard to read, perhaps break it down into smaller components.